main.cpp

```diff
@@ -2,24 +2,44 @@
 
 using namespace std;
 
+//struktury
+typedef struct{int x; int y;} COORDS;
+
 // zmienne globalne
 string mapa[100];
-int coordx, coordy;
-int endcoordx, endcoordy;
+COORDS player_coords;
+COORDS end_coords;
 char user='X';
 string file_name="plik";
 
 // procedury
 void getMap(); // wczytuje mapę z pliku file_name i zapisuje do mapa[]
+bool isExist(COORDS p) { // sprawdza czy dane pole jest na mapie
+  if(p.y>=SIZE)return false;
+  return mapa[p.y].size()>=p.x;
+}
+COORDS randomCoords(bool notWall=true) {
+  COORDS c;
+  do {
+    c.y = rand() % SIZE;
+    size_t s = mapa[c.y].size();
+    if(s) // s != 0
+      c.x = rand() % s;
+  }while(!(isExist(c)&&((!notWall)||isWall(c))));
```

Comment:
jak ktoś mi napisze jak to prościej napisać to mg poprawić, ale robiłem to na szybko i powinno działać xD